src/login_param.rs

```diff
@@ -25,19 +31,54 @@ impl Default for CertificateChecks {
     }
 }
 
+#[derive(Debug)]
+pub struct ServerParams {
+    pub protocol: Protocol,
+    pub socket: Socket,
+    pub hostname: String,
+    pub port: u16,
+    pub username_pattern: UsernamePattern,
+}
+
+pub type ImapServers = Vec<ServerParams>;
+pub type SmtpServers = Vec<ServerParams>;
+
+#[derive(Debug)]
+pub struct LoginParamNew {
```

Comment:
This name is unclear. Could it be named `AutoconfigParam` or something like this?

As I understand it, `loginparam_old_to_new` converts user-provided settings into the input for automatic configuration algorithm, and `loginparam_new_to_old` is a very simple autoconfiguration algorithm that converts results obtained from autoconfig XML or provider database into `LoginParam` with the first IMAP and SMTP server.

"Old" `LoginParam` is not going anywhere until automatic reconfiguration is implemented, and it is not a short-term goal. It is what is saved in the database under `configured_` settings.

Also this `LoginParamNew`, besides renaming, could be moved into `configure` module.

Reply:
Sure, I had not really thought about naming before and such things before.

Reply:
As for loginparam_old_to_new and the other way round, the thought was that I could implement point 5 only partly first and then already test it using these conversion methods. And that they might come handy later under a different name. 

So, feel free to rename and change these things, I never planned to merge these functions as they currently are.